tools/protoc-gen-docs/htmlGenerator.go

```diff
@@ -0,0 +1,805 @@
+	} else {
+		g.print("<h2 id='", name, "'>", name, "</h2>")
+	}
+	g.print("<section>")
+	g.print()
+}
+
+func (g *htmlGenerator) generateSectionTail() {
+	g.print("</section>")
+	g.print()
+}
+
+func (g *htmlGenerator) generateMessage(message *messageDescriptor) error {
+	g.generateSectionHeading(message.GetName())
+
+	if !g.printComments(message.path) {
+		g.warn(message.path, "no comment for message %s", message.GetName())
+	}
+
+	g.print()
+
+	g.print("<table>")
+	g.print("<tr>")
+	g.print("<th>Field</th>")
+	g.print("<th>Type</th>")
+	g.print("<th>Description</th>")
+	g.print("</tr>")
+
+	var oneof int32 = -1
+	for i, field := range message.Field {
+		fieldName := camelCase(*field.Name)
+		fieldTypeName, err := g.fieldTypeName(field)
+		if err != nil {
+			return err
+		}
+
+		if *field.Type == descriptor.FieldDescriptorProto_TYPE_MESSAGE {
+			o, err2 := g.findObject(field.GetTypeName())
+			if err2 != nil {
+				return err2
+			}
+
+			if d, ok := o.(*messageDescriptor); ok && d.GetOptions().GetMapEntry() {
+				keyType, err3 := g.fieldTypeName(d.Field[0])
+				if err != nil {
+					return err3
+				}
+
+				valType, err4 := g.fieldTypeName(d.Field[1])
+				if err4 != nil {
+					return err4
+				}
+
+				fieldTypeName = fmt.Sprintf("map&lt;%s, %s&gt;", keyType, valType)
+			}
+		}
+
+		if field.OneofIndex != nil {
+			class := ""
+			if field.Options != nil && *field.Options.Deprecated {
+				class = "deprecated "
+			}
+
+			if *field.OneofIndex != oneof {
+				class = class + "oneof oneof-start"
+				oneof = *field.OneofIndex
+			} else {
+				class = class + "oneof"
+			}
+			g.print("<tr class='", class, "'>")
+		} else {
+			if field.Options != nil && *field.Options.Deprecated {
+				g.print("<tr class='deprecated' title='Deprecated'>")
+			} else {
+				g.print("<tr>")
+			}
+		}
+
+		g.print("<td><code>", fieldName, "</code></td>")
+		g.print("<td><code>", fieldTypeName, "</code></td>")
+		g.print("<td>")
+
+		path := fmt.Sprintf("%s,%d,%d", message.path, messageFieldPath, i)
+		if !g.printComments(path) {
+			g.warn(path, "no comment for field %s", field.GetName())
+		}
+
+		g.print("</td>")
+		g.print("</tr>")
+	}
+	g.print("</table>")
+
+	g.generateSectionTail()
+
+	return nil
+}
+
+func (g *htmlGenerator) generateEnum(enum *enumDescriptor) {
+	g.generateSectionHeading(camelCase(enum.GetName()))
+
+	if !g.printComments(enum.path) {
+		g.warn(enum.path, "no comment for enum %s", enum.GetName())
+	}
+
+	g.print()
+
+	g.print("<table>")
+	g.print("<tr>")
+	g.print("<th>Name</th>")
+	g.print("<th>Description</th>")
+	g.print("</tr>")
+
+	for i, e := range enum.Value {
+		name := *e.Name
+
+		if e.Options != nil && *e.Options.Deprecated {
+			g.print("<tr class='deprecated' title='Deprecated'>")
+		} else {
+			g.print("<tr>")
+		}
+		g.print("<td><code>", name, "</code></td>")
+		g.print("<td>")
+
+		path := fmt.Sprintf("%s,%d,%d", enum.path, enumValuePath, i)
+		if !g.printComments(path) {
```

Comment:
consider throwing in the error message directly inside the generate's printComments function.. 
"Element has no comment". Any future code here will not have to worry about doing this if clause.